src/main/java/magicbees/itemInventories/InventoryBeeRing.java

```diff
@@ -0,0 +1,339 @@
+            parent.setTagCompound(nbt);
+        }
+
+        NBTTagCompound slotNbt;
+        if (!nbt.hasKey(KEY_SLOTS)) {
+            slotNbt = new NBTTagCompound();
+            nbt.setTag(KEY_SLOTS, slotNbt);
+        } else {
+            slotNbt = nbt.getCompoundTag(KEY_SLOTS);
+        }
+
+        String slotKey = getSlotNBTKey(slot);
+
+        if (stack == null) {
+            slotNbt.removeTag(slotKey);
+        } else {
+            NBTTagCompound itemNbt = new NBTTagCompound();
+            stack.writeToNBT(itemNbt);
+
+            slotNbt.setTag(slotKey, itemNbt);
+        }
+
+        setInventorySlot(parent);
+    }
+
+    // private method to manually set the inventory of a player
+    private void setInventorySlot(ItemStack parent) {
+
+        if (this.player.getHealth() == 0) {
+            return;
+        }
+
+        if (baubleFlag) {
+            InventoryBaubles baubles = PlayerHandler.getPlayerBaubles(this.player);
+            baubles.setInventorySlotContents(itemLocatedSlot, parent);
+        } else {
+            player.inventory.setInventorySlotContents(itemLocatedSlot, parent);
+        }
+    }
+
+    // Writes the effectData field to NBT in a readable way (This and the next method could probably be condensed, but
+    // I feel that this is the most readable form I could come up with.
+    public void writeEffectNBT() {
+        NBTTagCompound compound = parent.getTagCompound();
+
+        if (compound == null) {
+            compound = new NBTTagCompound();
+            parent.setTagCompound(compound);
+        }
+
+        NBTTagCompound slotNbt1;
+        if (!compound.hasKey(KEY_EFFECT_1)) {
+            slotNbt1 = new NBTTagCompound();
+            compound.setTag(KEY_EFFECT_1, slotNbt1);
+        } else {
+            slotNbt1 = compound.getCompoundTag(KEY_EFFECT_1);
+        }
+
+        NBTTagCompound slotNbt2;
+        if (!compound.hasKey(KEY_EFFECT_2)) {
+            slotNbt2 = new NBTTagCompound();
+            compound.setTag(KEY_EFFECT_2, slotNbt2);
+        } else {
+            slotNbt2 = compound.getCompoundTag(KEY_EFFECT_2);
+        }
+
+        for (int i = 0; i < 5; i++) {
+            String slotKey = getSlotNBTKey(i);
+            try {
+                slotNbt1.setInteger(slotKey + "i", effectData[0].getInteger(i));
+                slotNbt1.setBoolean(slotKey + "b", effectData[0].getBoolean(i));
+                slotNbt1.setFloat(slotKey + "f", effectData[0].getFloat(i));
+            } catch (Exception ignored) {}
+        }
+
+        for (int i = 0; i < 5; i++) {
+            String slotKey = getSlotNBTKey(i);
+            try {
+                slotNbt2.setInteger(slotKey + "i", effectData[1].getInteger(i));
+                slotNbt2.setBoolean(slotKey + "b", effectData[1].getBoolean(i));
+                slotNbt2.setFloat(slotKey + "f", effectData[1].getFloat(i));
+            } catch (Exception ignored) {}
+        }
+
+        setInventorySlot(parent);
```

Comment:
Please, explain why already equipped ring should be re-equipped? Do you want to update NBT?

Reply:
Yes, the NBT is updated here, so the slot must be updated. I was using references earlier to just edit the slot directly, but somewhere along the way the itemStack is copied so whatever was being passed wasn't actually the itemStack in the player's inventory.